rabbitmq_client/publisher.py

```diff
@@ -34,27 +36,51 @@ def register_serializer(self, func: Callable):
-            f"Expected instance of dict, passed {type(payload)}"
-        assert isinstance(headers, dict), \
-            f"Expected instance of dict, passed {type(headers)}"
+        assert self.serializer is not None, \
+            "Define custom serializer with @publish.register_serializer decorator"
+        assert isinstance(headers, dict), f"Expected instance of dict, passed {type(headers)}"
         assert isinstance(priority, int), \
             f"Expected instance of int, passed {type(priority)}"
 
-        serializer: Callable = self.serializer or default_serializer
-        stringified_payload = json.dumps(payload, default=serializer).encode('utf-8')
+        # Check that either payload or payload_list is provided, but not both None
+        if payload is None and (payload_list is None or len(payload_list) == 0):
+            raise ValueError("Either 'payload' or 'payload_list' must be provided.")
+
+        if payload_list is None or len(payload_list) == 0:
+            payload_list = [payload]
+
+        for payload_item in payload_list:
+            assert isinstance(payload_item, dict), \
+                f"Expected instance of dict, passed {type(payload_item)}"
+
+        self._publish_to_channel(queue_config, payload_list, headers, priority)
+
+    def _publish_to_channel(self, queue_config: PublishQueueConfig, payloads: List[dict],
+                            headers: Optional[dict] = None, priority: Optional[int] = 0):
         with get_connection(queue_config.broker_config) as connection:
             channel = connection.channel()
-            channel.basic_publish(exchange=queue_config.exchange,
-                                  routing_key=queue_config.routing_key,
-                                  body=stringified_payload,
-                                  properties=pika.BasicProperties(headers=headers, priority=priority))
+            channel.tx_select()
+            for payload in payloads:
+                stringified_payload = json.dumps(payload, default=self.serializer).encode('utf-8')
+                # ToDo: Fix this. Below queue_declare raises conflict if it(durable) does not match with definitions.json config
+                # channel.queue_declare(queue=queue_config.name)
+                try:
+                    channel.basic_publish(exchange=queue_config.exchange,
+                                          routing_key=queue_config.routing_key,
+                                          body=stringified_payload,
+                                          properties=pika.BasicProperties(headers=headers, priority=priority))
+                except Exception as e:
+                    logging.error(f"Got exception {e} while publishing message to queue {queue_config.routing_key}")
+                    raise PublisherException(f"Got exception {e} while publishing message to queue "
+                                             f"{queue_config.routing_key} for payload {payload}")
```

Comment:
if the idea is to log and reraise, then can do like this?
```suggestion
                    raise
```

Reply:
changed